updates

dk5144 opened the pull request.

dk5144 committed:
updates

thaddeuspearson commented on simpletix/events/templates/events/event_detail.html:
```diff
@@ -179,7 +185,21 @@ <h2 class="mb-1 text-dark fw-semibold">{{ event.title }}</h2>
+                          Organizers cannot buy
+                        </button>
+                      {% elif user.is_authenticated %}
+                        <a href="{% url 'orders:order' event.id %}"
```
this url will need the query param in order to pre-populate the correct ticket type:
`href="{% url 'orders:order' event.id %}"` -> `href="{% url 'orders:order' event.id %}?ticket_category_id={{ ticket.id }}"`

thaddeuspearson commented on simpletix/orders/views.py:
```diff
@@ -9,65 +6,150 @@
     event = get_object_or_404(Event, id=event_id)
-    preselect_ticket_category_id = request.GET.get("ticket_category_id", None)
-    if request.user and request.user.is_authenticated:
-        profile, _ = OrganizerProfile.objects.get_or_create(user=request.user)
```
by deleting this, the form does not have a `profile` arg passed to it, which means the users email is not automatically populated. I think there are a few approaches to fix this, but one might be to instantiate `profile` in a similar fashion that was here before, but specifically in the else clause (line 115) below.

thaddeuspearson commented on simpletix/orders/views.py:
```diff
@@ -9,65 +6,150 @@
+            except (ValueError, TicketInfo.DoesNotExist):
+                selected_ticket = None  # fall back to first ticket
+
+        form = OrderForm(event=event, initial=initial)
```
this form is expecting a `profile` arg. see my note about the deletion above.

thaddeuspearson reviewed (changes requested):
Hi @dk5144,

I think I was able to pinpoint why your branch seemed to be missing some features that we implemented. I should have caught them on the first review, so I sincerely apologize for the revert -> cherry-pick workflow you had to do for this PR. Completely my fault for rushing the review.

Let me know if you have any questions on my comments

dk5144 committed:
Updates

dk5144 commented:
Hi @thaddeuspearson ,

Thanks for the review comments! I’ve made the following updates based on your notes:

Added the ?ticket_category_id={{ ticket.id }} query param to all ticket-type links so the correct ticket option is pre-selected on the order page.

Restored the profile argument when initializing OrderForm, using the attendee’s UserProfile so the email field is auto-populated again as before.

Updated the GET branch in the order view to ensure profile is passed consistently, including when a ticket is pre-selected via query params.

These adjustments bring back the original behaviour without affecting any existing flows.

dk5144 committed:
resolved merge conflicts

dk5144 committed:
build fail fix

thaddeuspearson commented on simpletix/events/templates/events/event_detail.html:
@dk5144 I think this is probably what should remain for the ticket category tables

thaddeuspearson commented on simpletix/events/templates/events/event_detail.html:
```diff
@@ -238,8 +244,62 @@ <h2 class="mb-1 text-dark fw-semibold">{{ event.title }}</h2>
             </p>
           </div>
         {% endif %}
+        <table class="table align-middle">
```
@dk5144 I believe this is duplicated code for the ticket category, but there might have been a reason given the conditional logic. Can you confirm?

dk5144 replied:
That second table was duplicated ticket category markup. I’ve removed the extra table and consolidated everything into a single tickets table.

thaddeuspearson commented on simpletix/orders/views.py:
```diff
@@ -9,71 +6,154 @@
+    # Build profile (if any) so the form can pre-populate email
+    profile = None
+    if request.user.is_authenticated:
+        profile = UserProfile.objects.filter(user=request.user).first()
```
I think UserProfile does not have a `contact_email` field at this current moment. Check out the model here: 
https://github.com/gcivil-nyu-org/team1-mon-fall25/blob/504e00d222588cb7619261cad021b84fb5aee44b/simpletix/accounts/models.py#L12

dk5144 replied:
Good catch, you’re right, I switched this to use UserProfile but UserProfile doesn’t have contact_email, so the email can’t pre-populate.
For now I’ll change this back to use OrganizerProfile so the form keeps working, and we can track the larger UserProfile/OrganizerProfile refactor as a separate story.

thaddeuspearson reviewed (changes requested):
I left a few inline comments that need addressing. The email pre-population is still not working correctly, and we have a duplication of ticket category tables in the event detail template.

<img width="903" height="453" alt="Screenshot 2025-12-01 at 1 15 48 PM" src="https://github.com/user-attachments/assets/cb7a71fa-232c-4941-879a-a61bf4cef283" />


I think the issue with the email not pre-populating is because UserProfile does not have an email field that we set explicitly, but OrganizerProfile does.

This opens a slightly larger can of worms, as a refactor would likely be in order given how we currently handle the models for UserProfile and OrganizerProfile.

Maybe the quick fix here is to use the OrganizerProfile for the profile for the time being (like the way it was before) to get things working with the email pre-population (since [both an OrganizerProfile and a UserProfile](https://github.com/gcivil-nyu-org/team1-mon-fall25/blob/504e00d222588cb7619261cad021b84fb5aee44b/simpletix/accounts/views.py#L124) are created at signup), and then we can consider a larger refactor to create a single profile that has a role attribute, instead of separating everything at the model level

dk5144 committed:
resolve merge conflicts

dk5144 committed:
code fix

dk5144 commented:
@thaddeuspearson Thanks for checking! Yes, that section was a duplicate. I’ve removed the extra ticket-category block and verified the page still renders correctly across all conditions (sold-out, active tickets, organizer view, attendee view). Let me know if you'd like me to make any further changes.

thaddeuspearson reviewed (dismissed):
Changes look good @dk5144 ! I think there might be some merge conflicts that need addressing, but otherwise, LGTM!

dk5144 committed:
resolved merge conflicts

dk5144 commented:
@thaddeuspearson Can you pls check now? I have resolved the merge conflicts. 
Thanks!

thaddeuspearson commented on simpletix/events/templates/events/event_detail.html:
@dk5144 I think this section needs to be removed as a `Buy Ticket` button appears to be duplicated when logged in as an attendee

<img width="927" height="122" alt="Image" src="https://github.com/user-attachments/assets/e6be73c4-211b-41e9-bebc-e4b55661080a" />

thaddeuspearson commented on simpletix/events/templates/events/event_detail.html:
```diff
@@ -240,11 +260,53 @@ <h2 class="mb-1 text-dark fw-semibold">{{ event.title }}</h2>
+              {% endif %}
+            {% endif %}
+          {% else %}
+            {% if not event.is_cancelled %}
```
@dk5144 I think we also need to check if the event is sold out here. If there are no tickets remaining, there should not be a `Buy Ticket` button

dk5144 committed:
updated the fix for duplicate buy button

dk5144 commented:
@thaddeuspearson, sorry to miss that. I have updated the code. pls check if everything's looking good.
thanks.

thaddeuspearson reviewed (approved):
I think there are some opportunities to DRY this code up a bit (specifically we could reuse  `_user_owns_tickets` in a few other places), but given the deadline for the project is so close, I think this is good enough.

LTGM!
